File: src/main/java/me/lordsaad/modeoff/Modeoff.java
package me.lordsaad.modeoff;

import me.lordsaad.modeoff.common.CommandAssign;
import me.lordsaad.modeoff.common.CommandManager;
import me.lordsaad.modeoff.common.CommandTpPlot;
import me.lordsaad.modeoff.common.CommonProxy;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.SidedProxy;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.event.FMLServerStartingEvent;
import org.apache.logging.log4j.Logger;

@Mod(
		modid = Modeoff.MOD_ID,
		name = Modeoff.MOD_NAME,
		version = Modeoff.VERSION,
		dependencies = "required-after:librarianlib"
)
public class Modeoff {

	public static final String MOD_ID = "modeoff";
	public static final String MOD_NAME = "Modeoff";
	public static final String VERSION = "1.0";

	public static final String CLIENT = "me.lordsaad.modeoff.client.ClientProxy";
	public static final String SERVER = "me.lordsaad.modeoff.server.ServerProxy";

	public static Logger logger;

	@SidedProxy(clientSide = CLIENT, serverSide = SERVER)
	public static CommonProxy proxy;

	@Mod.Instance
	public static Modeoff instance;

	@Mod.EventHandler
	public void preInit(FMLPreInitializationEvent event) {
		proxy.preInit(event);
		logger = event.getModLog();
	}

	@Mod.EventHandler
	public void init(FMLInitializationEvent e) {
		proxy.init(e);
	}

	@Mod.EventHandler
	public void postInit(FMLPostInitializationEvent e) {
		proxy.postInit(e);
	}

	@Mod.EventHandler
	public void serverLoad(FMLServerStartingEvent event) {
		event.registerServerCommand(new CommandAssign());
		event.registerServerCommand(new CommandTpPlot());
		event.registerServerCommand(new CommandManager());
	}
}

File: src/main/kotlin/me/lordsaad/modeoff/Modeoff.kt
package me.lordsaad.modeoff

import me.lordsaad.modeoff.common.CommonProxy
import me.lordsaad.modeoff.common.command.CommandAssign
import me.lordsaad.modeoff.common.command.CommandManager
import me.lordsaad.modeoff.common.command.CommandRank
import me.lordsaad.modeoff.common.command.CommandTpPlot
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.SidedProxy
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import net.minecraftforge.fml.common.event.FMLServerStartingEvent
import org.apache.logging.log4j.Logger

@Mod(
    modid = Modeoff.MOD_ID,
    name = Modeoff.MOD_NAME,
    version = Modeoff.VERSION,
    dependencies = "required-after:librarianlib",
    modLanguageAdapter = "net.shadowfacts.forgelin.KotlinAdapter"
)
object Modeoff {
    const val MOD_ID = "modeoff"
    const val MOD_NAME = "Modeoff"
    const val VERSION = "1.0"

    const val CLIENT = "me.lordsaad.modeoff.client.ClientProxy"
    const val SERVER = "me.lordsaad.modeoff.server.ServerProxy"

    lateinit var logger: Logger private set

    @SidedProxy(clientSide = CLIENT, serverSide = SERVER)
    lateinit var proxy: CommonProxy private set

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        proxy.preInit(event)
        logger = event.modLog
    }

    @Mod.EventHandler
    fun init(e: FMLInitializationEvent) {
        proxy.init(e)
    }

    @Mod.EventHandler
    fun postInit(e: FMLPostInitializationEvent) {
        proxy.postInit(e)
    }

    @Mod.EventHandler
    fun serverLoad(event: FMLServerStartingEvent) {
        event.registerServerCommand(CommandAssign())
        event.registerServerCommand(CommandTpPlot())
        event.registerServerCommand(CommandManager())
        event.registerServerCommand(CommandRank())
    }
}
